pkg/eks/fargate.go

```diff
@@ -64,12 +65,15 @@ func DoCreateFargateProfiles(config *api.ClusterConfig, fargateClient FargateCli
 		//
 		// In the case that a ResourceInUseException is thrown on a profile which was
 		// created on an earlier call, we do not error but continue to the next one
+		var e *eks.ResourceInUseException
 		err := fargateClient.CreateProfile(profile, true)
-		switch errors.Cause(err).(type) {
-		case nil:
+		switch {
+		case err == nil:
 			logger.Info("created Fargate profile %q on EKS cluster %q", profile.Name, clusterName)
-		case *eks.ResourceInUseException:
+		case errors.As(err, &e):
 			logger.Info("Either Fargate profile %q already exists on EKS cluster %q or another profile is being created/deleted, no action taken", profile.Name, clusterName)
+		case fargate.IsUnauthorizedError(err):
+			return errors.Wrapf(err, "either account is not authorized to use Fargate or region %s is not supported", config.Metadata.Region)
```

Comment:
big nit: can we add a unit test :smile:

Reply:
yep. we do have a test which check for failing errors (unlike an other in this switch which would not fail only log), so it is kindof covered, but i can add something explicit for this error

Reply:
the only difference here is that we alter the error message